user-service/user-domain/src/main/kotlin/com/grepp/quizy/domain/user/User.kt

```diff
@@ -0,0 +1,12 @@
+package com.grepp.quizy.domain.user
+
+import java.time.LocalDateTime
+
+
+class User (
+    val id: UserId,
+    val userProfile: UserProfile,
+    val role: Role = Role.USER,
+    val provider: ProviderType,
```

Comment:
private 써주는게 좋지않을까요 ?
```suggestion
    private val id: UserId,
    private val userProfile: UserProfile,
    private val role: Role = Role.USER,
    private val provider: ProviderType,
```

Reply:
넵 private 붙이고 캡슐화 진행해보겠습니다..!

Reply:
전부 private을 붙이면 다른 곳에서 단일 프로퍼티 값만 필요할 때 접근이 안 되지 않나요?
예를 들어 토큰 발급 등에서 `id`만 필요할 때 따로 함수를 만드는 것보단 이정도는 그냥 getter로도 괜찮지 않나 싶어서용

Reply:
수빈님 말씀은 `user.id` 이걸 못한다 이런뜻인가요?

근데 이게 코틀린이라서 조금 확신이없긴한데 어제 지원님이 말씀하신 것 처럼 원래는 getter를 사용하지 않도록 하는게 좋습니다. 
그래서 자바에서도 원래 엔티티를 만들 때 모두 private로 하고 getter를 써야할 때는 getter를 만들어서쓰는데 귀찮으니 롬복의 `@Getter`를 사용하져

최대한 캡슐화를 지키기 위해서는 모두 private로 막아두고, 필요한 프로퍼티에 대해서만 getter를 이용하는게 좋다고 생각합니다. 이렇게 하는게 뭔가 코틀린스럽지않은 것 같긴한데 val로 설정해두고 정말 변경이 필요할 때만 var로 바꾸는거처럼 모두 private로 막아두고 객체에 메시지를 보내는 방식이 아닌 값을 꺼내서 사용해야만 할 때가 되었을 때 getter를 정의하는게 캡슐화를 지키기 좋은 습관입니다.

물론 저도 잘 못지킵니다 화이팅 ㅎㅎ

Reply:
넵 저도 민혁님 말씀이 맞는것같네요
어제 말씀해주신 내용대로라면 private하게 둬서 함부로 접근하지 못하는게 좋아보여요
일단 최대한 모든 필드는 private으로 두고 진행해보겠습니다!

Reply:
아하 이해했습니다!
처음엔 전부 private으로 두고 추후에 꼭 그 프로퍼티 값이 필요한 코드가 생기면
그때 public으로 돌려서 getter를 쓸 수 있게 하는 거 맞나요?

그게 맞겠네용 저도 지금 하던 거 그렇게 적용하겠습니다!

Reply:
제 생각에는 Public으로 돌리는건 좋지않은 것 같구요 private인데 커스텀 프로퍼티를 만들 수 있기 때문에 getter를 재정의하는게 좋지 않을까? 라고 생각하기는 합니다 !
필드가 public으로 존재하는 순간 객체의 캡슐화의 의미가 사라진다고 생각하기는 해서 getter를 만드는게 좋아보여요

근데 이게 저도 생각이 자바에서 코틀린으로 변환하는 과정이있어서... 근데 코틀린을 잘 몰라서 ㅋㅋㅋ ㅠㅠ

그냥 자바로 보면 private이지만 getter를 사용해야 할 때 getter를 만든다 라고 생각하면 좋을 것 같아요
그리고 생각보다 getter를 막아두게 되면 getter를 먼저 만들 생각보다, 외부세계(객체를 사용하려는쪽)보다는 내부세계(객체 내부)로 처리 할 수 있는 방법이없을까? 를 고민하게 된다고 생각해요.
객체지향 패러다임에 중요한 부분이 객체간에 메시지를 전달하는거라고 생각하거든요 getter가 없다면 객체 내부의 값을 함부로 쓰는게 아니라 객체에게 메시지를 전달할 방법을 고민 할 수 있게 된다고 생각해요

클린코드 책에서 나오는 디미터의 법칙이라고 있는데 이것도 한번 알아보시면 좋을 것 같아여
근데 중요한 것은 저는 완전 자바기반으로 생각해서 코틀린에서는 다를수도 있기 때문에 코틀린스럽게 하는 방법이 있다면 알려주시면 저도 반영하겠습니다

Reply:
음 그쳐 자바 관점에선 민혁님 말씀에 완전 동의합니다!

커스텀 getter는 만약 getter에서 반환하는 값이 프로퍼티 값에 변형을 줘야 하는 거면 재정의가 필요한 게 맞는데
단순 프로퍼티 값 자체만 반환하기 위한 목적이면 굳이 getter를 따로 정의할 필요는 없을 거 같아요!
예를 들어 `getId() = id` 이런거용

코틀린 자체가 자바에서 `@Getter` 등의 보일러 플레이트를 줄이기 위해 프로퍼티 형식으로 설계된 거라고 알고 있었습니다!
`user.id` 이런 식으로 접근해도 어차피 내부 필드 값은 private이고 실제 호출은 getter로 접근하도록 되어 있으니까용